examples/Nano33SenseSend/Nano33SenseSend.ino

```diff
@@ -0,0 +1,204 @@
+/**-----------------------------------------------------------------------------
+
+    Example code to send data using Arduino Nano 33 Sense board, an
+    Adafruit MAX98357A and a mini oval speaker
+
+    @file Nano33SenseSend.ino
+
+    @brief Create a developer account at https://developers.chirp.io,
+    and copy and paste your key, secret and config string for the "arduino"
+    protocol into the credentials.h file.
+
+    This example will start listening for chirps and print to the terminal
+    when anything is received.
+
+    Circuit:
+      - Arduino Nano 33 BLE board
+      - Adafruit MAX98357A amplifier
+      - Adafruit Mini Oval speaker
+
+    Copyright © 2011-2019, Asio Ltd.
+    All rights reserved.
+
+  ----------------------------------------------------------------------------*/
+#include "chirp_connect.h"
+#include "credentials.h"
+
+#define NUM_BUFFERS            8
+#define BUFFER_SIZE            1024
+#define OUTPUT_SAMPLE_RATE     16667
+
+#define I2S_SCK_PIN            23  // D7
+#define I2S_DATA_PIN           21  // D8
+#define I2S_LRCK_PIN           27  // D9
+
+// Global variables ------------------------------------------------------------
+
+static chirp_connect_t *chirp = NULL;
+int16_t data[NUM_BUFFERS][BUFFER_SIZE];
+uint8_t next, current;
+
+// Function definitions --------------------------------------------------------
+
+void chirpErrorHandler(chirp_connect_error_code_t code);
+void setupChirp(void);
+void i2s_init(void);
+void i2s_start(void);
+
+// Main ------------------------------------------------------------------------
+
+void setup()
+{
+  current = 0;
+
+  Serial.begin(115200);
+  while(!Serial);  // Wait for Serial monitor before continuing
+
+  setupChirp();
+  sendRandomChirp();
+
+  i2s_init();
+  i2s_start();
+}
+
+void loop()
+{
+  if (NRF_I2S->EVENTS_TXPTRUPD != 0)
+  {
+    // Transfer is complete, start it with the preprocessed data
+    current = (current + 1) % NUM_BUFFERS;
+    NRF_I2S->TXD.PTR = (uint32_t)(data[current]);
+    NRF_I2S->EVENTS_TXPTRUPD = 0;
+
+    // Process next buffer
+    next = (current + 1) % NUM_BUFFERS;
+    chirp_connect_error_code_t err = chirp_connect_process_shorts_output(chirp, data[next], BUFFER_SIZE);
+    chirpErrorHandler(err);
+  }
+}
+
+// Chirp -----------------------------------------------------------------------
+
+void
+onSendingCallback(void *chirp, uint8_t *payload, size_t length, uint8_t channel)
+{
+  Serial.println("Sending data...");
+}
+
+ void
+onSentCallback(void *chirp, uint8_t *payload, size_t length, uint8_t channel)
```

Comment:
A detail but could we have function declaration on one single line ?

Reply:
done. good to merge